clustered/client/src/main/java/org/ehcache/clustered/client/internal/service/DefaultClusteringService.java

```diff
@@ -337,8 +337,18 @@ public String toString() {
   }
 
   private static ExecutorService createAsyncWorker() {
+    ThreadGroup initialGroup = Thread.currentThread().getThreadGroup();
     return Executors.newSingleThreadExecutor(r -> {
-      Thread t = new Thread(r, "Async DefaultClusteringService Worker");
+      ThreadGroup group = initialGroup;
+      while (group != null && group.isDestroyed()) {
+        ThreadGroup parent = group.getParent();
+        if (parent == null) {
+          break;
+        } else {
+          group = parent;
+        }
+      }
+      Thread t = new Thread(group, r, "Async DefaultClusteringService Worker");
```

Comment:
Is this racy?  What happens if thread group is destroyed between looping and thread creation?

Reply:
Yes, it's racy... but in a worse way than you think. An unstarted thread won't prevent a thread group from being destroyed. So yes I could catch `IllegalThreadStateException` here and continue the loop of trying the parents... but I can't prevent the group from being destroyed between when I create the thread and when the executor starts the actual thread. Since use of thread groups is rare, and when it does happen they are usually (should be?) application context scoped (and therefore lifecycled alongside the cache?) I think we're as safe as we need to be. If you want me to I can narrow the race in the way you describe though.